### PR TITLE
Allow optional src_path argument to generate that default to URL

### LINKED ISSUE #146: project is read-only
### Description

I ran 

```julia
using COPIERTemplate
COPIERTemplate.generate("YourPackage.jl")
```

The file permissions are read-only, I expected it to use my umask.

* YourPackage/ has `drwxr-xr-x`
*  YourPackage/README.md has `-r--r--r--` 
*  YourPackage/src has  `drwxr-xr-x`
*  YourPackage/test/Project.toml `-r--r--r--`

### Package Version

0.3.1

### Julia Version

1.10

### Reproduction steps

1. I installed Julia using juliaup
2. Start console with `~/.juliaup/bin/julia`
3. Ran command from description
4. The permissions of files in YourPackage/ are not correct

When I use copier it self it has correct permssions

```
python3 -m venv venv
. venv/bin/activate
pip install copier
copier copy https://github.com/abelsiqueira/COPIERTemplate.jl YourPackage.jl
```
Then the permissions are user writable (`-rw-r--r--`) as I expected.

I am running on Windows inside WSL on a Ubuntu 22.04 .

I fixed the permissions with
```
find YourPackage.jl -type f -exec chmod u+w {} \;
```

### Relevant log output

_No response_

### Operating System

Linux

### Checks

- [X] There is not an [existing issue](https://github.com/abelsiqueira/COPIERTemplate.jl/issues) with the same question
- [X] I have read the [contributing guide](https://github.com/abelsiqueira/COPIERTemplate.jl/blob/main/docs/src/contributing.md)
- [X] I agree with the [code of conduct](https://github.com/abelsiqueira/COPIERTemplates.jl/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by sverhoeven**
Instead of using the copier.yml from ~/.julia/** use GH url.

TODO
* [x] make url an argument
* [x] use same branch/repo for template as where generate() function is defined, To make sure in a PR that changes both generator and template are tested together. 

## Related issues

Fixes #146

## Checklist

- [x] I am following the [contributing guidelines](https://github.com/abelsiqueira/COPIERTemplate.jl/blob/main/docs/src/90-contributing.md)
- [x] Tests are passing
- [x] Lint workflow is passing
- [x] Docs were updated and workflow is passing
- [x] [CHANGELOG.md](https://github.com/abelsiqueira/COPIERTemplate.jl/blob/main/CHANGELOG.md) was updated

### CODE REVIEW

**Comment by abelsiqueira** [comment]
Zenodo failures are 50x.

**Comment by abelsiqueira** [comment]
@sverhoeven I'm not sure I understood your second bullet point. This is testing the generation of the template using wrapper (Julia) vs direct (copier) in two situations: using the source "URL#main", and using the source "dir of cloned/added package".

**Comment by abelsiqueira** [comment]
@sverhoeven would you like to review, or should I ask @suvayu?

**Comment by suvayu** [comment]
> I'm not sure I understood your second bullet point.

In the original commit the URL was hard coded, so if you opened a PR in this repo, CI would still run on a template url that points to main, not the branch in the PR.

In the test, if you are passing a fixed url (haven't looked at your commit yet), then it should be passing the url for the PR branch for the test to be correct.

**Review by suvayu — CHANGES_REQUESTED** [review]
The test needs to address the 2nd bullet.

**Review by suvayu — APPROVED** [review]

File: test/runtests.jl [hunk 1]
```diff
@@ -51,6 +51,18 @@ end
 
 bash_args = vcat([["-d"; "$k=$v"] for (k, v) in template_options]...)
 template_path = joinpath(@__DIR__, "..")
+template_url = "https://github.com/abelsiqueira/COPIERTemplate.jl"
```
[the remark this answers is not in the document]
Comment (abelsiqueira) [reply test/runtests.jl:54]: Explained offline

**Comment by abelsiqueira** [comment]
Explained offline that this actually addresses the second bullet point and the PR was approved then.

**Comment by abelsiqueira** [comment]
@allcontributors add @suvayu for review

**Pull request merged**